App/src/Repository/GroupRepository.php

```diff
@@ -6,29 +6,43 @@
 use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
 use Doctrine\Persistence\ManagerRegistry;
 
+/**
+ * @extends ServiceEntityRepository<Group>
+ */
 class GroupRepository extends ServiceEntityRepository
 {
     public function __construct(ManagerRegistry $registry)
     {
         parent::__construct($registry, Group::class);
     }
 
-    public function findIdByName(string $name): int
+    public function findIdByName(string $name): ?int
     {
-        return $this->createQueryBuilder('g')
+        $result = $this->createQueryBuilder('g')
             ->select('g.id')
             ->where('g.name = :name')
             ->setParameter('name', $name)
             ->getQuery()
             ->getSingleScalarResult();
+
+        if (!is_int($result)) {
+            return null;
+        }
+        return $result;
     }
 
     public function findOneById(int $id): ?Group
     {
-        return $this->createQueryBuilder('g')
+        $group = $this->createQueryBuilder('g')
             ->where('g.id = :id')
             ->setParameter('id', $id)
             ->getQuery()
             ->getOneOrNullResult();
+
+        if ($group !== null && !$group instanceof Group) {
```

Comment:
Écrit de la sorte, je trouve que ta condition est plus intelligible en lecture

```suggestion
        if ($group === null || !$group instanceof Group) {
```

Reply:
done